packages/runtime/src/signal/signal.ts

```diff
@@ -1,26 +1,28 @@
 import deepEqual from 'fast-deep-equal'
 
-const signals: Set<Signal<any>> = ((window as any).currentSignals =
-  (window as any).currentSignals ?? new Set<Signal<any>>())
-
 export class Signal<T> {
   value: T
-  subscribers: Array<{
+  subscribers: Set<{
     notify: (value: T) => void
     destroy?: () => void
   }>
   subscriptions: Array<() => void>
 
   constructor(value: T) {
     this.value = value
-    this.subscribers = []
+    this.subscribers = new Set()
     this.subscriptions = []
-    signals?.add(this)
   }
   get() {
     return this.value
   }
   set(value: T) {
+    // Short circuit and skip expensive `deepEqual` if there are not currently any subscribers
+    if (this.subscribers.size === 0) {
+      this.value = value
+      return
+    }
+
     if (deepEqual(value, this.value) === false) {
       this.value = value
       this.subscribers.forEach(({ notify }) => notify(this.value))
```

Comment:
I guess this would be similar?
```suggestion
    this.value = value
    // Short circuit and skip expensive `deepEqual` if there are not currently any subscribers
    if (this.subscribers.size === 0) {
      return
    }

    if (deepEqual(value, this.value) === false) {
      this.subscribers.forEach(({ notify }) => notify(this.value))
```